lib/matplotlib/axes/_axes.py

```diff
@@ -3112,6 +3118,74 @@ def boxplot(self, x, notch=False, sym=None, vert=True, whis=1.5,
                             if ci[1] is not None:
                                 stats['cihi'] = ci[1]
 
+        if notch is False:
```

Comment:
`notch` needs to default to `None` and then this test should be `if notch is None:`.  As written it is impossible to over-ride a `True` rcparam value of `notch` with a `False` value by the user.